data_model/src/predicate.rs

```diff
@@ -576,29 +689,92 @@ pub mod numerical {
             Ok(())
         }
 
+        #[test]
+        fn interval_semantics_u32() {
+            use Value::U32;
+            {
+                let pred = Range::U32((1_u32, 100_u32).into());
+                println!("semi_interval range predicate: {pred:?}");
+
+                assert!(pred.applies(&U32(1_u32)));
+                assert!(!pred.applies(&U32(0_u32)));
+                assert!(pred.applies(&U32(100_u32)));
+                assert!(!pred.applies(&U32(101_u32)));
+            }
+            {
+                let pred = Range::U32((127_u32, 127_u32).into());
+                assert!(pred.applies(&U32(127_u32)));
+                assert!(!pred.applies(&U32(126_u32)));
+                assert!(!pred.applies(&U32(128_u32)));
+            }
+        }
+
+        #[test]
+        fn interval_semantics_u128() {
+            use Value::U128;
+            let pred = Range::U128((1_u128, 100_u128).into());
+
+            assert!(pred.applies(&U128(1_u128)));
+            assert!(!pred.applies(&U128(0_u128)));
+            assert!(pred.applies(&U128(100_u128)));
+            assert!(!pred.applies(&U128(101_u128)));
+        }
+
+        #[test]
+        #[allow(clippy::panic_in_result_fn)] // ? for syntax simplicity.
+        fn interval_semantics_fixed() -> Result<(), fixed::FixedPointOperationError> {
+            let pred = Range::Fixed((Fixed::try_from(1_f64)?, Fixed::try_from(100_f64)?).into());
+
+            assert!(pred.applies(&Value::Fixed(Fixed::try_from(1_f64)?)));
+            assert!(!pred.applies(&Value::Fixed(Fixed::try_from(0.99_f64)?)));
+            assert!(pred.applies(&Value::Fixed(Fixed::try_from(99.9999_f64)?)));
```

Comment:
Why not
```suggestion
            assert!(pred.applies(&Value::Fixed(Fixed::try_from(100_f64)?)));
```
?

Reply:
There's a fundamental difference between Binary coded decimal (which would be exact) and binary fixed-point, so we have to test this.

Reply:
```suggestion
            assert!(pred.applies(&Value::Fixed(Fixed::try_from(99.999_999_999_9_f64)?)));
```
to differentiate from `semi_interval_semantics_fixed()`